kv/union_store_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"bytes"
+
+	. "github.com/pingcap/check"
+)
+
+var _ = Suite(&testUnionStoreSuite{})
+
+type testUnionStoreSuite struct {
+	k1, k2, k3, k4 []byte
+	v1, v2, v3, v4 []byte
+
+	store MemBuffer
+	us    UnionStore
+}
+
+func (s *testUnionStoreSuite) SetUpTest(c *C) {
+	s.k1 = []byte("k1")
+	s.k2 = []byte("k2")
+	s.k3 = []byte("k3")
+	s.k4 = []byte("k4")
+	s.v1 = []byte("1")
+	s.v2 = []byte("2")
+	s.v3 = []byte("3")
+	s.v4 = []byte("4")
+
+	s.store = NewMemDbBuffer()
+	s.us = NewUnionStore(&mockSnapshot{s.store})
+}
+
+func (s *testUnionStoreSuite) TearDownTest(c *C) {
+	s.us.Close()
+}
+
+func (s *testUnionStoreSuite) TestGetSet(c *C) {
+	s.store.Set(s.k1, s.v1)
+	v, err := s.us.Get(s.k1)
+	c.Assert(err, IsNil)
+	c.Assert(bytes.Compare(v, s.v1), Equals, 0)
+	s.us.Set(s.k1, s.v2)
+	v, err = s.us.Get(s.k1)
+	c.Assert(err, IsNil)
+	c.Assert(bytes.Compare(v, s.v2), Equals, 0)
+}
+
+func (s *testUnionStoreSuite) TestDelete(c *C) {
+	s.store.Set(s.k1, s.v1)
+	err := s.us.Delete(s.k1)
+	c.Assert(err, IsNil)
+	_, err = s.us.Get(s.k1)
+	c.Assert(IsErrNotFound(err), IsTrue)
```

Comment:
I think we may implement a `Checker` in `terror` package. Then we are able to do something like `c.Assert(err, terror.Equal, kv.ErrNotExist)`. @siddontang @qiuyesuifeng